packages/iTwinUI-react/src/core/Header/HeaderSplitButton.tsx

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { DropdownMenu } from '../DropdownMenu';
+import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
+import SvgCaretUpSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretUpSmall';
+
+import { PolymorphicForwardRefComponent, useTheme } from '../utils';
+import { SplitButtonProps } from '../Buttons';
+
+export type HeaderSplitButtonProps = SplitButtonProps;
+
+type HeaderSplitButtonComponent = PolymorphicForwardRefComponent<
+  'button',
+  HeaderSplitButtonProps
+>;
+
+export const HeaderSplitButton: HeaderSplitButtonComponent = React.forwardRef(
+  (props, forwardedRef) => {
+    const {
+      onClick,
+      menuItems,
+      className,
+      menuPlacement = 'bottom-end',
+      size,
+      children,
+      style,
+      title,
+      type = 'button',
+      startIcon,
+      ...rest
+    } = props;
+
+    useTheme();
+
+    const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+
+    const [menuWidth, setMenuWidth] = React.useState(0);
+    const ref = React.useRef<HTMLLIElement>(null);
+
+    React.useEffect(() => {
+      if (ref.current) {
+        setMenuWidth(ref.current.offsetWidth);
+      }
+    }, [children, size]);
```

Comment:
I dont think we should react to `size` prop change... actually, changing button size should not do anything.

Reply:
This is probably applicable to line 36 of `HeaderDropdownButton` also.

Reply:
I have removed the `size` prop as it is not needed for any `HeaderButton` type. Reducing the size of the button can be done by passing `isSlim` into `Header`. The previous `HeaderButton` did not have a `size` prop, so this should not cause any breaking changes